core/ops_builtin.rs

```diff
@@ -129,18 +129,21 @@ pub async fn op_void_async_deferred() {}
 /// Remove a resource from the resource table.
 #[op2(core, fast)]
 pub fn op_close(
-  state: &mut OpState,
+  state: Rc<RefCell<OpState>>,
   #[smi] rid: ResourceId,
 ) -> Result<(), Error> {
-  state.resource_table.close(rid)?;
+  let resource = state.borrow_mut().resource_table.take_any(rid)?;
+  resource.close();
   Ok(())
 }
 
 /// Try to remove a resource from the resource table. If there is no resource
 /// with the specified `rid`, this is a no-op.
 #[op2(core, fast)]
-pub fn op_try_close(state: &mut OpState, #[smi] rid: ResourceId) {
-  let _ = state.resource_table.close(rid);
```

Comment:
We should deprecate this method -- it's definitely a potential footgun.

Reply:
How is it a footgun? I use it all over the place in `ext/node`

Reply:
Sorry, I mean `resource_table.close`